src/modm/platform/spi/rp/spi_master_dma_impl.hpp.in

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2022, Andrey Kunitsyn
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_RP_SPI_MASTER{{ id }}_DMA_HPP
+#	error 	"Don't include this file directly, use 'spi_master_{{ id }}_dma.hpp' instead!"
+#endif
```

Comment:
This is an old dumb pattern of us, that can be replaced with
```cpp
#pragma once
#include "spi_master_{{ id }}_dma.hpp"
```
That way code completion also works correctly again in this file in most IDEs.